### ISSUE TITLE
[PNI] Using backspace in mobile search is very delayed

### ISSUE THREAD

**Issue by tbrlpld** [mozilla/foundation.mozilla.org#9821]
### Describe the bug

In the mobile search bar on the PNI homepage, when using backspace it takes a long time for the text to actually disappear. The problem seems to get worse the more often the key is used. 

It seems like this has something to do with the debounce.

### To Reproduce

1. Go to https://foundation.mozilla.org/en/privacynotincluded/ with an iPhone.
2. Click on the search icon.
3. Type something, e.g. "Germin".
4. Try to correct your typo by backspacing.
5. Notice that it takes a long time for the text to disappear.
6. Once the text does disappear, try typing new text, e.g. "Garmin"
7. This will also not appear immediately character-by-character, but take some time and then appear all at once.

### Expected behavior

When hitting the backspace the text should disappear immediately.

### Screenshots

Here is a screen recording. In times where it seems like nothing is happening, I am typing or trying to delete... That is the issue.

https://user-images.githubusercontent.com/24797493/206480747-c198e9f6-cb34-431d-bedf-eb16c304be4b.MP4

### Smartphone (please complete the following information):
 - Device: iPhone 14 Pro
 - OS: iOS 
 - Browser: Safari (but does not matter)
 - Version: 16

**Comment by mmmavis** [comment]
I'm curious if this is also happening on Android/Windows phones.

**Comment by tbrlpld** [comment]
😱 Are Windows phones a thing again?

**Comment by mmmavis** [comment]
hahah I knew someone is gonna say something about it. I was trying to be inclusive-ish lol

**Comment by fessehaye** [comment]
@beccaklam can you test staging on your mobile and see if this issue has gotten better or worse?

**Comment by tbrlpld** [comment]
The tying delay has gotten better, but I am experiencing some weirdness with the visibility of the search bar now. Seems like the search bar is sometimes hidden or blocked by the dropdown when scrolling back up. Also, it sometimes takes quite some time for the grid items to show again. 


https://user-images.githubusercontent.com/24797493/213828314-353b449b-d1f8-4012-ae6e-e01d8fe9a479.MP4

**Comment by beccaklam** [comment]
@fessehaye Hi! Sorry just testing this now. The backspace is working better but I'm also experiencing all the bugs that Tibor mentioned above?

**Comment by tbrlpld** [comment]
@beccaklam Do you remember is these issues where an issue before the backspace fix? I don't remember, but it might have been because I was distracted with the backspace issue. 

Question is if this is ok to go out as is with a follow or if we need a fix before we deploy to production?

**Comment by cdanfon** [comment]
@kristinashu Can you make a call about whether we can push to prod and address the issues that Tibor experienced whilst testing in a new ticket, or if they need to be fixed before we push to prod?

**Comment by kristinashu** [comment]
I just checked the "new" bug on production and I see it there as well so this isn't really a new issue.

The use of backspace on staging is way better now so please go ahead and push this to prod!

I can open a new ticket for the newly found bug.

**Comment by kristinashu** [comment]
Here's the new ticket for the hidden search bar https://github.com/mozilla/foundation.mozilla.org/issues/10051

@tbrlpld you also mentioned "Also, it sometimes takes quite some time for the grid items to show again." but I'm not seeing this. Do you want to open a ticket for that?

**Comment by tbrlpld** [comment]
Thanks for creating the follow up @kristinashu If you don't experience the grid load issue, I am happy to leave it out until it resurfaces again.
